### ISSUE TITLE
Add ability to compile and download a Linux VST

### ISSUE THREAD

**Issue by jrdooley** [enzienaudio/heavy#101]
The ability to download a Linux VST binary, or provide a template makefile in the source download would be a welcomed enhancement.

**Comment by diplojocus** [comment]
@jrdooley yep we should this! Good idea. We already have linux makefiles in the unity target so we should be able to apply that to VSTs.

**Comment by Simon-L** [comment]
I absolutely +1 on this! 

@diplojocus : Are these Makefiles for the Windows or the Linux (though still experimental) build of the Unity Editor? Last time I checked I think project files for Windows only were provided.

**Comment by diplojocus** [comment]
When you download the `Unity Src` target you'll see something like the following:

<img width="338" alt="screen shot 2016-07-07 at 16 06 37" src="https://cloud.githubusercontent.com/assets/594588/16658085/e77db228-445c-11e6-9184-dcf7f03a678f.png">

There's all the source files, android and linux/unix makefiles, xcode and vs2015 projects.

For the Editor side of things, the plugins are more dependent on the C# script which should work across all platforms. Then it's a case of telling Unity which binary to use for your development machine and for all the other platform/build settings. Windows makefiles (VS2015 project) definitely works as we use that for the `Unity Win32` and `Unity Win64` targets. Admittedly I haven't actually managed to test the linux beta yet.

**Comment by mhroth** [comment]
@jrdooley Are you looking for a 32 or 64-bit build for VSTs? Both? We'd compile on Ubuntu 14, does that work for you?

**Comment by jrdooley** [comment]
@mhroth : 64-bit on Ubuntu 14 works for me. My current systems are both 64-bit--Ubuntu 14 and Linux Mint 18(Ubuntu 16)--and I'd be using the VSTs with Bitwig (only works on 64-bit systems, though does support 32-bit VSTs) and possibly Ardour.

**Comment by mhroth** [comment]
@jrdooley Does this file work for you? You may need to rename the file to `libheavy.so`, I'm not sure what filename your DAW will look for. [heavy.so.zip](https://github.com/enzienaudio/heavy/files/370343/heavy.so.zip)

**Comment by mhroth** [comment]
A `Makefile` has been added to the vst2 generated files (in the `unix` directory). It the build product remains untested and therefore will not be exposed on the website. But the file is there in case anyone wants to try it out.

**Comment by jrdooley** [comment]
@mhroth: I've test the compiled VST you attached above and it works without any problems; I didn't even have to rename it as you suggested.

I've just created a new project on my Heavy user area and downloaded the source code, though I can't find a directory name `unix`. How do I access the `Makefile`?

**Comment by mhroth** [comment]
Thanks for testing @jrdooley! So, can you confirm that the name of the unix VST should be e.g. `heavy.so` and _not_ `libheavy.so`. Does it matter?

This functionality is currently in the `dev` release, which you can access using the [uploader script](https://enzienaudio.com/docs/index.html#3.uploader) and supplying the `--gen vst2` and `--release dev` flags.

**Comment by jrdooley** [comment]
To confirm, both `heavy.so` and `libheavy.so` work--the name of the file doesn't appear to matter. This has been tested in Bitwig Studio.

I'm struggling to get the uploader script to work. I run the script with the following arguments and get the following error:
`formuls@formuls-mint ~/Desktop $ python2 uploader.py /home/formuls/Desktop/basic/ --out /home/formuls/Desktop/compiled/ --name basic --gen vst2 -b --release dev`

`Enter username: formuls`

`Enter password:`

`Error: strptime() argument 1 must be string, not None
Getting a weird error? Get the latest uploader at https://enzienaudio.com/static/uploader.py`

I've also tried it without the `-b` argument and still get the same error.

**Comment by mhroth** [comment]
@jrdooley I've updated the uploader script. Try downloading the [newest version](https://enzienaudio.com/static/uploader.py) and giving it another try. Your command line is correct, the `-b` argument isn't particularly needed.

**Comment by jrdooley** [comment]
@mhroth The script successfully uploads a Pd patch and downloads source files, including VS2015 and XCode files. However, there is no `Makefile`, nor is there a `unix` directory, so no Linux VSTs :-(

**Comment by jrdooley** [comment]
@mhroth : I noticed that this issue has been marked solved. However, when I upload a patch with the uploader script using the `--gen vst2` and `--release dev` flags, the resulting source code that is downloaded does not contain a `unix` directory nor a `Makefile`.

**Comment by mhroth** [comment]
@jrdooley I'm sorry about that. Not sure what happened there, but _now_ it should be fixed.

**Comment by jrdooley** [comment]
@mhroth : Wicked! Just successfully downloaded and compiled a Linux VST. Nice one!
